packages/core/src/entities/App.ts

```diff
@@ -1,60 +1,114 @@
 // [ ] (ipfs) htmlUrl 	String 	The HTTP URL of the app HTML page. Uses the IPFS HTTP provider. E.g. http://gateway.ipfs.io/ipfs/QmdLEDDfi…/index.html
 // [ ] (ipfs) contentUrl 	String 	The HTTP URL of the app content. Uses the IPFS HTTP provider. E.g. http://gateway.ipfs.io/ipfs/QmdLEDDfi…/
-// [ ] (ipfs) icons 	{ src: String, sizes: String }[] 	Array of icons for the app (follows the web app manifest icons format).
-// [x] name 	String 	Name of the app, from the repository. E.g. "Tokens".
-// [x] registryAddress 	String 	Address of the aragonPM registry for this app.
-// [-] (need name prop) registry 	String 	Name of the aragonPM registry for this app. E.g. "aragonpm.eth"
-// [ ] (ipfs) sourceUrl 	String 	URL of the app source code.
-// [x] version 	String 	The current version of the app.
-// [x] kernelAddress 	String 	The address of the kernel.
-// [x] isForwarder 	Boolean 	Whether the app can act as a forwarder.
+// [ ] registry 	String 	Name of the aragonPM registry for this app. E.g. "aragonpm.eth"
 // [ ] tags 	String[] 	Tags associated with the app.
-// [x] App#repo()
-// [ ] App#abi()
-// [ ] App#intents()
-// [ ] App#deprecatedIntents()
 
 export interface AppData {
-  name?: string
   address: string
   appId: string
-  version?: string
-  registryAddress: string
+  artifact?: string | null
+  codeAddress: string
+  contentUri?: string
+  isForwarder?: boolean | null
+  isUpgradeable?: boolean | null
   kernelAddress: string
-  isForwarder: boolean
+  manifest?: string | null
+  name?: string
+  registryAddress: string
+  repoAddress?: string
+  version?: string
 }
 
 export default class App extends Entity implements AppData {
-  readonly name?: string
+  readonly abi?: Abi
   readonly address!: string
   readonly appId!: string
-  readonly version?: string
-  readonly registryAddress!: string
+  readonly appName?: string
+  readonly author?: string
+  readonly chainId?: AragonEnvironment
+  readonly codeAddress!: string
+  readonly contentUri?: string
+  readonly contentUrl?: string
+  readonly contractPath?: string
+  readonly description?: string
+  readonly htmlPath?: string
+  readonly htmlUrl?: string
+  readonly intents?: AppIntent[]
+  readonly deprecatedIntents?: { [version: string]: AppIntent[] }
+  readonly icons?: { src: string; sizes: string }[]
+  readonly isForwarder?: boolean | null
+  readonly isUpgradeable?: boolean | null
   readonly kernelAddress!: string
-  readonly isForwarder!: boolean
+  readonly name?: string
+  readonly registryAddress!: string
+  readonly registry?: string
+  readonly repoAddress?: string
+  readonly sourceUrl?: string
+  readonly tags?: string[]
+  readonly version?: string
 
-  constructor(data: AppData, connector: ConnectorInterface) {
+  constructor(
+    { artifact, manifest, ...data }: AppData,
+    connector: ConnectorInterface
+  ) {
     super(connector)
 
+    // TODO: If no metadata, fallback to resolve ourselves with ipfs
```

Comment:
What you think of having the IPFS fallback logic for every connector, here?

Reply:
Should we implement this at the connector level, as it seems specific to the TheGraph connector? We could pass a global IPFS URL resolver to every connector, so they can do anything they want with it.

Or should we make the IPFS provider connector-specific? We could imagine connectors that wouldn’t need any IPFS provider.

Reply:
We should discuss this live imo. When would things fall back? Which values would be affected?

Reply:
Well actually is going to be the Ethereum Connector that will need to use the fallback logic the most, that's why I suggest doing it at the core library level instead of giving that responsibility to the connectors. 

>When would things fall back?

When the data we receive don't have an artifact.json or manifest.json we try to resolve it ourselves using the contetUri.